[screen-capture] feat: add ability to listen for screenshots (#9747)
expo/expo: docs/pages/versions/unversioned/sdk/screen-capture.md

### Example: functions

```javascript
import * as ScreenCapture from 'expo-screen-capture';
import React from 'react';
import { Button, View } from 'react-native';

export default class ScreenCaptureExample extends React.Component {
  componentDidMount() {
    /* @info The provided function will be run whenever a screenshot of your app is taken. */

>     ScreenCapture.addScreenShotListener(() => {

Reviewer: ```suggestion
    ScreenCapture.addScreenshotListener(() => {
```

      alert('Thanks for screenshotting my beautiful app 😊');
    }); /* @end */
  }

  render() {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Button onPress={this._activate}>Activate</Button>
        <Button onPress={this._deactivate}>Deactivate</Button>

>         <Button onPress={this._listen}>Listen for screenshots</Button>

Reviewer 2: Hm looks like this` _listen` function isn't defined

Author: removed 👍

      </View>
    );
  }

  _activate = async () => {
    /* @info Screen will be uncapturable once <strong>preventScreenCaptureAsync()</strong> is called. */
    await ScreenCapture.preventScreenCaptureAsync(); /* @end */
  };

  _deactivate = async () => {
    /* @info Re-allows screen capture, or does nothing if preventScreenCaptureAsync() was never called. */
    await ScreenCapture.allowScreenCaptureAsync(); /* @end */
  };
}
```

## API

```js
import * as ScreenCapture from 'expo-screen-capture';
```

> <TableOfContentSection title='Methods' contents={['usepreventscreencapture(key)', 'preventScreenCaptureAsync(key)', 'allowScreenCaptureAsync(key)', 'addScreenShotListener(listener)', 'removeScreenShotListener(subscription)']} />

Reviewer: ```suggestion
<TableOfContentSection title='Methods' contents={['usepreventscreencapture(key)', 'preventScreenCaptureAsync(key)', 'allowScreenCaptureAsync(key)', 'addScreenshotListener(listener)', 'removeScreenshotListener(subscription)']} />
```

- **key (string)** [Optional] If provided, this will help prevent multiple instances of the `preventScreenCaptureAsync` and `allowScreenCaptureAsync` methods (and `usePreventScreenCapture` hook) from conflicting with each other. When using multiple keys, you'll have to re-allow each one in order to re-enable screen capturing. Defaults to `default`.

### `allowScreenCaptureAsync(key)`

Re-allows the user to screen record or screenshot your app. If you haven't called `preventScreenCaptureAsync()` yet, this method does nothing

#### Arguments

- **key (string)** [Optional] The value must be the same as the `key` passed to `preventScreenCaptureAsync` in order to re-enable screen capturing.

> ### `addScreenShotListener(listener)`

Reviewer: Shouldn't it be `addScreenshotListener`?

Adds a listener that will fire whenever the user takes a screenshot while the app is foregrounded. On Android, this method requires the `READ_EXTERNAL_STORAGE` permission- you can request this with [`Permissions.askAsync(Permissions.CAMERA_ROLL)`](../sdk/permissions/#permissionscamera_roll).

#### Arguments

- **listener (function)** The function that will be executed when the user takes a screenshot. This function accepts no arguments.

#### Returns

A `Subscription` object that you can use to unregister the listener, either by calling `.remove()` or passing it to `removeScreenShotListener`.

### `removeScreenShotListener(subscription)`

> Removes the subscription you provide, so that you are no longer listening for screen shots.

Reviewer: ```suggestion
Removes the subscription you provide, so that you are no longer listening for screenshots.
```
❓
